### DISCUSSION TITLE
Getting error while running docker build or using client command

### DISCUSSION THREAD

**Discussion by TahirAnny** [foyzulkarim/mernboilerplate-antd#45] [Q&A] [answered]
Does anyone have any idea why I am getting this error? I am kinda stuck here, I have added [ant pro utils from here](https://www.npmjs.com/package/@ant-design/pro-utils), but still getting this error in both the ways of running the project.
![image](https://user-images.githubusercontent.com/35254801/156871010-19c54745-8c60-4599-9b14-790449fe245f.png)

I tried to watch a few videos of this playlist, as I am not a regular follower of this playlist, kinda jumping from the middle, I know I am missing something but can't figure it out, so any help or suggestion will be really helpful.

**Comment by ArnobChowdhury** [comment]
Relatively new issue in ant design pro components. Follow this thread for possible solution https://github.com/ant-design/pro-components/issues/4772

**Reply by foyzulkarim** [reply]
Can you kindly describe the reproducing steps?

**Reply by TahirAnny** [reply]
After cloning the repo, I had created a new branch. Then I thought of running the application using the [visual studio code](https://github.com/foyzulkarim/mern-boilerplate#visual-studio-code). 
 * First, I have created a .env file and copy-pasted from [.env.sample](https://github.com/foyzulkarim/mern-boilerplate/blob/main/server/.env.sample). **Note** first time I actually missed replacing MONGODB_CLOUD_URL, then I re-watched [this video](https://www.youtube.com/watch?v=C8QiKCKbdqM&list=PLEYpvDF6qy8aUl1KnB1oaZbwLE2oPcZmz&index=49) and get that thing right.
 * Then I used server [command from](https://github.com/foyzulkarim/mern-boilerplate#server-commands) commands. One more thing, after running `npm run db:up` this command, I opened a new terminal in VS code and then ran `npm start` and it was giving me the below result
![image](https://user-images.githubusercontent.com/35254801/156886192-54433fe1-a8c3-4dae-a0c0-3f17b03c6573.png)

 *  After that, I moved to the client and typed `yarn` and that's when I got the above error, I thought maybe I don't have `ant pro` added, that's why I am getting the error. Then I search a bit and then added [ant pro utils from here](https://www.npmjs.com/package/@ant-design/pro-utils). After that, whenever I was running yarn, I was getting the below output
![image](https://user-images.githubusercontent.com/35254801/156886251-35c29e45-9abc-4c77-869a-74ed5f0c0070.png)

I thought, maybe I am missing something and watched [this video as well](https://www.youtube.com/watch?v=czNRCMLa0nk&list=PLEYpvDF6qy8aUl1KnB1oaZbwLE2oPcZmz&index=22)

As nothing was working, I thought let's use [Docker](https://github.com/foyzulkarim/mern-boilerplate#docker). I changed everything mentioned in this [note](https://github.com/foyzulkarim/mern-boilerplate#notes). Next, I ran `docker-compose build` command but it's giving me this error again
![image](https://user-images.githubusercontent.com/35254801/156886485-642aaef6-fdfe-4262-9b8f-66026d73e864.png)

I was curious to know what happens if I use `docker-compose -f docker-compose.mongocloud.yml build` command, and then the same thing was happening.

My apology, In case If I miss something that you mentioned and for this lengthy description. 

**Note:** every time I cloned the application and tried different ways.

**Reply by foyzulkarim** [reply]
I will prepare a video by following the readme steps.
I can guarantee if you follow readme file properly, your app should be up and running without any problem. 
I guess you didn't follow the readme properly. Please correct me if I am wrong 🙂

**Reply by foyzulkarim** [reply]
Also try to use node lts version to run the client

**Reply by foyzulkarim** [reply]
@TahirAnny please try to follow the readme steps and let me know if any step is confusing or not working. If needed I will do a call + screen sharing session with you to solve your issue. thanks.

**Reply by TahirAnny** [reply]
I will try again and definitely let you know InShaAllah

**Reply by TahirAnny** [reply]
Well, it's working fine now. 

Two things I have changed, 

1. Mongo DB container name
2. Node version 16.14.0 (I was using 17.1.0) (this because I was getting another error)

Though the error I was getting doesn't make any sense with these changes. It might happen, the issue Arnob Bhaia had referred, that's resolved now. 

Thank you.

**Comment by foyzulkarim** [comment, marked as answer]
For those who face the problem, please use `nodejs lts` version.
